packages/@lwc/engine/src/framework/decorators/register.ts

```diff
@@ -185,7 +184,7 @@ export function registerDecorators(
     if (!isUndefined(wire)) {
         for (const fieldOrMethodName in wire) {
             const { adapter, method } = wire[fieldOrMethodName];
-            const { configCallback } = wire[fieldOrMethodName];
+            const configCallback = wire[fieldOrMethodName].config;
```

Comment:
you can still use destructuring here: `const { config: configCallback } = wire[fieldOrMethodName];`

Reply:
Or just

```
for (const {adapter, method, config: configCallback} of wire) {
```

Reply:
@rtm we have issues with the `for of` (performance issues) on the compiled code for old browsers (IE11), that's why you will see no `for of` in our platform code until that's solved.

Reply:
Could you provide a reference or background for this? In the TS playground, ``for (const a of []) {}` emits `for (var _i = 0, _a = []; _i < _a.length; _i++) {  var a = _a[_i]; }` (when targeting ES5).

Reply:
@rtm that's the wrong transformation for the semantics of the iterator protocol, that is just a typescript optimization because it knows that you are passing an array directly, but if the value is unknown for the compiler, it has to implement the full semantics.

Btw, we do not compile LWC library code with typescript, we do it with babel. This is an example of the transformation: https://babeljs.io/en/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=GYewTgBAFAxiB2BnALhA3gQwCYYA7IFMwAaCAWwOQAsQtS55gBLAcwC4IHmWBhDAG34AjDDADWAXwghgEAO5MwBAJToJQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.5.5&externalPlugins=